Home page and footer along with responsiveness

sulattphone opened the pull request:
Code for:
 - Home page: index.hbs
 - Footer: partials/footer.hbs
 - CSS: <s>assets/css/app.css</s> src/sass/app.scss

Wilson's header doesn't show up on my local instance.  Not sure if that's an issue or not.

sulattphone committed:
footer first and stats bubbles

sulattphone committed:
Merge branch 'master' of https://github.com/IEEEBerkeley/ieee-ghost into theme-SuLatt

sulattphone committed:
Revert "Merge branch 'master' of https://github.com/IEEEBerkeley/ieee-ghost into theme-SuLatt"

This reverts commit 9c9a0e526ea4c1b98d2602f6164dfb339b022160, reversing
changes made to 74ab78087c62187a470867bce602e06f3b0602b8.

sulattphone committed:
home page fixed and done

sulattphone committed:
home page responsiveness

sulattphone committed:
Footer done with responsivesness

sulattphone committed:
added my css at the end of app.scss

PotatoParser commented on partials/footer.hbs:
```diff
@@ -3,27 +3,96 @@ This footer template is shared across all the pages.
+          <a href="#" class="footer-link">Check-In</a><br />
+          <a href="#" class="footer-link">About Us</a><br />
+          <a href="#" class="footer-link">Sponsors</a><br />
+          <a href="#" class="footer-link">Resources</a>
```
Marking as todo (adding the links) :D

PotatoParser commented on partials/footer.hbs:
```diff
@@ -3,27 +3,96 @@ This footer template is shared across all the pages.
+      <p>
+          <a href="#" class="footer-link">Blog</a><br />
+          <a href="#" class="footer-link">Check In</a><br />
+          <a href="#" class="footer-link">Execs and Officers</a><br />
```
Todo: add links

sulattphone replied:
Oh yeah, I wasn't sure how to add those links actually, because I don't think I can use an absolute web address there right?

PotatoParser replied:
Oh! By using a forward slash, you reference the link from the website (relative linking):
For instance, if your website was `https://site.com/`
`href="/tothispage"`
is the same as
`href="https://site.com/tothispage"`

You just have to add in `href="/blog/<add the page link here>"`

encadyma replied:
Would be careful about prepending `/blog`, since this will eventually be in production.

I think this is done better through the Ghost interface. There is a "secondary navigation" option that Ghost provides, which could be repurposed for footer links. I found [some documentation here](https://ghost.org/docs/api/v3/handlebars-themes/helpers/navigation/), but from what I can tell this would be similar to how the top header gets links from Ghost.

PotatoParser replied:
Whoops! Kevin is totally correct here. It should be:
`href="{{ @site.url }}/<route to your page>"`

sulattphone replied:
Oh yeah I noticed that while actually fixing this. Thanks!

PotatoParser commented on index.hbs:
```diff
@@ -76,11 +76,85 @@ into the {body} of the default.hbs template --}}
+                    <div class="three-columns stats-columns">
+                      <div class="stats-bubbles">
+                        <span class="stats">500 General Members</span>
+                      </div>
```
The ellipses should be a circle:
```html
<div class="circle-text">
    <p>500</p>
    <p>General</p>
    <p>Members</p>
<div>
```

```scss
.circle-text {
  padding: 30px;
  border-radius: 3000px;
  text-align: center;
  background-color: blue;
  display: inline-block; // This can be ignored if you use a container that has "display: flex;"
  color: white;
  p {
    margin: 0;
  }
}
```

PotatoParser commented on index.hbs:
```diff
@@ -76,11 +76,85 @@ into the {body} of the default.hbs template --}}
+
+                <div class="join-div div-spacing">
+                  <h2 class="ieee-blue">Join Our Community!</h4>
+                  <a href="/apply" type="button" class="ieee-btn link"><span class="btn-text"> Apply Now</span></a>
```
There should be a hover selector so that the box is noticeable when you hover over it.

PotatoParser commented on partials/footer.hbs:
```diff
@@ -3,27 +3,96 @@ This footer template is shared across all the pages.
+    <div class="four-columns footer-columns">
+      <div class="footer-brand-div text-center">
+        <h2>University of California, Berkeley IEEE Student Branch</h2>
+        <p>251 CoryHall, University of California, Berkeley</p>
```
Fix space between `CoryHall` and use full address (with zip code, etc.)

PotatoParser commented on src/README.md:
```diff
@@ -7,38 +7,6 @@ install all dependencies with `npm install`, then launch the environment with
 `npm run docker-watch`. Webpack will watch changes made to the repository and
 update the theme (+ refresh your browser) accordingly.
 
```
Not sure why this is being removed. It might be because the new branch was missing the README.

encadyma replied:
Seems to be the case, in this case I would probably pull in changes from README.md that Wilson made

sulattphone replied:
You mean just pull the README.md in src file from master? Or is it from a branch Wilson made?

PotatoParser replied:
I think from my branch, though I would be careful of conflicts we may have between our branches.

PotatoParser commented on partials/footer.hbs:
```diff
@@ -3,27 +3,96 @@ This footer template is shared across all the pages.
+        {{/if}}
+      </nav>
+     
+      <a href="/apply" type="button" class="ieee-btn link"><span class="footer-btn-txt">Join Our Mailing List</span></a>
```
There should be a different link here and there should be a hover animation.

encadyma replied:
I would probably give a little more space between the mailing list button and the social media buttons

PotatoParser commented on partials/footer.hbs:
```diff
@@ -3,27 +3,96 @@ This footer template is shared across all the pages.
+    </div>
+  </div>
+  <div class="row copyright-row text-center">
+    <p class="">@Copyright 2020 IEEE Student Branch at University of California, Berkeley</p>
```
Fix typo: `© 2020 IEEE Student Branch at University of California, Berkeley`

PotatoParser commented on src/sass/app.scss:
```diff
@@ -69,3 +69,125 @@
+  
+  .stats {
+      color: white;
+      font-size: 1em;
```
The text is a little small. I think you could up this value.

PotatoParser reviewed (changes requested):
Overall, looking very nice!
There seems to be some visual issues with the circles. (Showing up as ovals and breaking when it comes to responsiveness)
![image](https://user-images.githubusercontent.com/45542237/82277743-7ef10180-993d-11ea-81b6-060cf7269301.png)

encadyma commented on uc-berkeley-ieee.ghost.2020-05-18-22-42-22.json:
```diff
@@ -0,0 +1 @@
+{"db":[{"meta":{"exported_on":1589841742106,"version":"3.13.4"},"data":{"actions":[{"id":"5e40d4f90f6e325a8d9d093e","resource_id":"1","resource_type":"user","actor_id":"1","actor_type":"user","event":"edited","context":null,"created_at":"2020-02-10T03:58:49.000Z"},{"id":"5e40f52b0f6e325a8d9d0944","resource_id":"5e40d5630f6e325a8d9d093f","resource_type":"user","actor_id":"1","actor_type":"user","e ...
```
This seems to be sensitive information from your personal instance of Ghost, I would probably `git rm` this so it doesn't show up in the branch.

PotatoParser replied:
Should a force push be used here? It seems that this JSON file is from our active Ghost website.

sulattphone replied:
Oh yeah this is the content of our website. I just thought I would include it.. idk why lol. I can delete it.

encadyma commented on src/sass/app.scss:
```diff
@@ -69,3 +69,125 @@
 @import "libs/aos/aos";
+
+
+/************ custome theme ***********************/
```
some inconsistent tabbing in the file

encadyma commented on src/package-lock.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "liebling",
+  "name": "ieee-ghost",
```
Whoops, forgot to update the name, thanks for fixing this.

encadyma commented on partials/footer.hbs:
```diff
@@ -3,27 +3,96 @@ This footer template is shared across all the pages.
+      <a href="/apply" type="button" class="ieee-btn link"><span class="footer-btn-txt">Join Our Mailing List</span></a>
+    </div>
+  </div>
+  <div class="row copyright-row text-center">
```
On mobile, the copyright text takes up the full width, while the other elements of the footer are centered or padded. I think a little padding wouldn't hurt.

encadyma commented on index.hbs:
```diff
@@ -76,11 +76,85 @@ into the {body} of the default.hbs template --}}
+                </div>  {{!-- End of paragraph-div --}}
+
+
+                <div class="stats-div text-center div-spacing">
```
Looking back at the original design, the circle design is hard to pull off while looking nice, and I think a far more effective design choice that other websites use a lot is keeping the proportions of the original markup, but removing the circle and coloring the number and the subtitle text differently. I made a mockup of what that might look like: 

![mockup](https://i.fluffy.cc/WlG8s72r2JpccKj8q1Q0Hqv5LvCJQ7Z1.png)

PotatoParser replied:
Woah, that font looks really nice. Could you link the font used here?

encadyma reviewed (changes requested):
Looks good, just a couple of changes. Thanks for the PR!

sulattphone committed:
Changes wilson suggested

sulattphone committed:
Made changes from Kevin

andrewkdinh committed:
Create footer

encadyma reviewed (approved):
Changes look good!

PotatoParser commented on src/sass/app.scss:
```diff
@@ -69,3 +69,143 @@
+	font-size: 1.5em;
+	border-radius: 0.3rem;
+  padding: 1rem;
+  border: 1px solid black;
```
I think removing this line (`border: 1px solid black;`) makes the buttons look nicer :)

PotatoParser commented on src/sass/app.scss:
```diff
@@ -69,3 +69,143 @@
+  border: 1px solid black;
+}
+
+.ieee-btn:hover {
```
I feel that this hover animation is a bit off. I made a couple of example animations here: [https://codepen.io/PotatoParser/pen/MWaZexv](https://codepen.io/PotatoParser/pen/MWaZexv). I think making the box have a shadow and make it lighter looks nicer. (I also added a transition duration to make things more smooth).

andrewkdinh committed:
Footer fixes

- Correct copyright
- Change relative links to absolute
- Add hover animations for links

PotatoParser commented on src/sass/app.scss:
```diff
@@ -69,3 +69,143 @@
+.footer-link:hover {
+  color: #166bbb;
+}
+
```
I think there should be a transition duration
```css
.footer-link {
    transition: color 0.2s;
}
```
(The complete style for `footer-link` in scss):
```scss
.footer-link {
    transition: color 0.2s;
    &:hover {
            color: #166bbb;
    }
}
```

PotatoParser reviewed (changes requested):
Looking nicer every commit! I just have some visual ideas and fixes.

sulattphone committed:
Changes to link transitions and btn animation

sulattphone committed:
copyright text fix

sulattphone committed:
Fixing button display on firefox

andrewkdinh committed:
Added IEEE logo

Added IEEE logo to middle of footer. Logo dissapears on mobile

PotatoParser reviewed (approved):
Things look nice, though I think that some additional changes should be made:
+ Font should change in the area with the 500+ Members, etc.
+ Underline beneath the 500+ Members, etc, should be less wide on mobile devices.

Overall, very nice PR!

andrewkdinh committed:
Merge branch 'footer' into theme-SuLatt

andrewkdinh committed:
Finished merging footer branch

Add address, email, JS license, and headers for links

sulattphone committed:
Merge branch 'theme-SuLatt' of https://github.com/IEEEBerkeley/ieee-ghost into theme-SuLatt

sulattphone committed:
footer finalized
